tests/test_memory_load.py

```
@@ -0,0 +1,256 @@
+import tempfile
+from pathlib import Path
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+
+# ---------------------------------------------------------------------------
+# Helpers
+# ---------------------------------------------------------------------------
+
+def _make_middleware(knowledge_store=None):
```

Ruff (0.15.10)

[warning] Missing return type annotation for private function `_make_middleware`

(ANN202)